distributed/scheduler.py

```diff
@@ -1778,6 +1778,7 @@ def __init__(
         self.task_groups = {}
         self.task_prefixes = {}
         self.task_metadata = {}
+        self.total_memory = 0
```

Comment:
It's almost overkill to maintain additional state for this but if we have to call `identity` often I want it to be constant time

Reply:
Also, I'm surprised we are not tracking this already. Seems like a useful metric for adaptive scaling or smth like that 🤷